### PR TITLE
feat(config): make wallet application and paths configurable

### PR DESCRIPTION
**Pull request by Phil91**
## Description

* make application of the wallet configurable
* make wallet paths configurable

## Why

To be able to support multiple wallets

## Issue

Refs: #226

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/ssi-credential-issuer/blob/main/docs/technical-documentation/dev-process/How%20to%20contribute.md)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added copyright and license headers, footers (for .md files) or files (for images)

### CODE REVIEW

**Review by evegufy — CHANGES_REQUESTED** [review]

File: src/externalservices/Wallet.Service/DependencyInjection/WalletSettings.cs [hunk 1]
```diff
@@ -33,4 +33,19 @@ public class WalletSettings : BasicAuthSettings
 
     [Required]
     public int EncryptionConfigIndex { get; set; }
+
+    [Required]
+    public string WalletApplication { get; set; } = null!;
+
+    [Required]
+    public string CredentialCreationPath { get; set; } = null!;
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/externalservices/Wallet.Service/DependencyInjection/WalletSettings.cs]: adjusted

File: src/externalservices/Wallet.Service/DependencyInjection/WalletSettings.cs [hunk 2]
```diff
@@ -33,4 +33,19 @@ public class WalletSettings : BasicAuthSettings
...
+
+    [Required]
+    public string SigningCredentialPath { get; set; } = null!;
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/externalservices/Wallet.Service/DependencyInjection/WalletSettings.cs]: adjusted

File: charts/ssi-credential-issuer/values.yaml [hunk 3]
```diff
@@ -124,6 +124,16 @@ processesworker:
...
     # -- Client-secret for wallet client-id. Secret-key 'wallet-client-secret'.
     clientSecret: ""
+    # -- the application set in the wallet
+    application: "catena-x-portal"
+    # -- path to create a credential
+    credentialCreationPath: "api/v2.0.0/credentials"
+    # -- path to sign a specific credential {0} will be replaced by the credential id
+    signingCreationPath: "/api/v2.0.0/credentials/{0}"
+    # -- path to get a specific credential {0} will be replaced by the credential id
+    getCreationPath: "/api/v2.0.0/credentials/{0}"
```
[the remark this answers is not in the document]
Comment (Phil91) [reply charts/ssi-credential-issuer/values.yaml]: adjusted

File: charts/ssi-credential-issuer/values.yaml [hunk 4]
```diff
@@ -124,6 +124,16 @@ processesworker:
...
+    # -- path to revoke a specific credential {0} will be replaced by the credential id
```
[the remark this answers is not in the document]
Comment (Phil91) [reply charts/ssi-credential-issuer/values.yaml]: adjusted

File: charts/ssi-credential-issuer/values.yaml [hunk 5]
```diff
@@ -124,6 +124,16 @@ processesworker:
     clientId: "wallet-client-id"
     # -- Client-secret for wallet client-id. Secret-key 'wallet-client-secret'.
     clientSecret: ""
+    # -- the application set in the wallet
+    application: "catena-x-portal"
+    # -- path to create a credential
+    credentialCreationPath: "api/v2.0.0/credentials"
+    # -- path to sign a specific credential {0} will be replaced by the credential id
+    signingCreationPath: "/api/v2.0.0/credentials/{0}"
+    # -- path to get a specific credential {0} will be replaced by the credential id
```
[the remark this answers is not in the document]
Comment (Phil91) [reply charts/ssi-credential-issuer/values.yaml]: adjusted

File: charts/ssi-credential-issuer/values.yaml [hunk 6]
```diff
@@ -124,6 +124,16 @@ processesworker:
     clientId: "wallet-client-id"
     # -- Client-secret for wallet client-id. Secret-key 'wallet-client-secret'.
     clientSecret: ""
+    # -- the application set in the wallet
+    application: "catena-x-portal"
+    # -- path to create a credential
+    credentialCreationPath: "api/v2.0.0/credentials"
+    # -- path to sign a specific credential {0} will be replaced by the credential id
```
[the remark this answers is not in the document]
Comment (Phil91) [reply charts/ssi-credential-issuer/values.yaml]: adjusted

File: charts/ssi-credential-issuer/values.yaml [hunk 7]
```diff
@@ -124,6 +124,16 @@ processesworker:
...
+    signingCreationPath: "/api/v2.0.0/credentials/{0}"
```
[the remark this answers is not in the document]
Comment (Phil91) [reply charts/ssi-credential-issuer/values.yaml]: adjusted

File: src/processes/Processes.Worker/appsettings.json [hunk 8]
```diff
@@ -47,6 +47,11 @@
     "TokenAddress": "",
     "BaseAddress": "",
     "EncryptionConfigIndex": 0,
-    "EncryptionConfigs": []
+    "EncryptionConfigs": [],
+    "WalletApplication": "",
+    "CreateCredentialPath": "",
+    "SigningCredentialPath": "",
```
Comment (evegufy) [inline src/processes/Processes.Worker/appsettings.json:53]:
```suggestion
    "SignCredentialPath": "",
```

**Review by evegufy — CHANGES_REQUESTED** [review]

**Pull request merged**